polygeist/lib/Conversion/PolygeistToLLVM/PolygeistToLLVM.cpp

```diff
@@ -1210,6 +1210,34 @@ struct ConvertPolygeistToLLVMPass
       LLVM_DEBUG(llvm::dbgs() << "ConvertPolygeistToLLVMPass: Module after:\n";
                  m->dump(); llvm::dbgs() << "\n";);
     }
+
+    if (!options.useGenericFunctions) {
+      // During conversion, duplicate malloc and free functions can be generated
+      // if the user calls either of these functions and also uses `new` while
+      // using "non-generic" allocation functions. Remove potential duplicates
+      // here.
+      constexpr StringLiteral freeFn("free");
+      constexpr StringLiteral mallocFn("malloc");
+      std::array<std::pair<StringRef, bool>, 2> functionFound{
+          std::pair<StringRef, bool>{freeFn, false},
+          std::pair<StringRef, bool>{mallocFn, false}};
+      std::vector<LLVM::LLVMFuncOp> toRemove;
+      for (auto func : m.getOps<LLVM::LLVMFuncOp>()) {
+        StringRef name = func.getName();
+        auto *iter = llvm::find_if(functionFound,
```

Comment:
can user create specialized function with name `free` or `malloc`?

Reply:
Yes, that will lead to an error now. This PR improves current behavior:
1. When using `free`/`malloc` with correct signature: now it will never lead to error (circumvented by using `llvm.func`, something I'm avoiding in #11248);
2. When a bad signature is used, this now yields a comprehensible error.

Reply:
Are we trying to only remove declarations? What if the `LLVMFuncOp` are definitions with different bodies?